src/aktualizr_primary/main.cc

```diff
@@ -122,11 +122,14 @@ int main(int argc, char *argv[]) {
     conn = aktualizr.SetSignalHandler(f_cb);
 
     if (!config.uptane.secondary_config_file.empty()) {
-      if (boost::filesystem::exists(config.uptane.secondary_config_file)) {
+      try {
         Primary::initSecondaries(aktualizr, config.uptane.secondary_config_file);
-      } else {
-        LOG_WARNING << "The specified secondary config file does not exist: " << config.uptane.secondary_config_file
-                    << "\nProceed further without secondary(ies)";
+      } catch (const std::exception &e) {
+        LOG_ERROR << "Secondary initialization failed";
```

Comment:
I would add e.what() to the log message so we see more details of the secondary provisioning failure.

Reply:
In the current version, `e.what()` is logged in `initSecondaries()` but I agree this is maybe not really clear.
`initSecondaries()` could also log these errors and return a boolean, to simplify the call site in `main.cc`.

Reply:
I see, makes sense, it's up to you whether to terminate exceptions in initSecondaries and return boolean or rethrow them.